### PR TITLE
[CoSim] py-version updates for SDof update

### PR DESCRIPTION
**Pull request by philbucher**
in an nutshell now the vars are real attributes of the Modules, also the CoSim-module can now have its own variables

needed for #5837

### CODE REVIEW

**Comment by philbucher** [comment]
Ping @adityaghantasala

File: applications/CoSimulationApplication/custom_data_structure/pyKratos/Variable.py [hunk 1]
```diff
@@ -44,55 +46,32 @@ def __str__(self):
...
+def CreateComponentVariable(module, name, source_variable, component_index):
+    if name in KM._registered_variables:
         raise NameError('Variable "{}" exists already!'.format(name))
-    globals()[name] = VariableComponent(name, source_variable, component_index)
+    KM._registered_variables[name] = VariableComponent(name, source_variable, component_index)
+    setattr(module, name, KM._registered_variables[name])
 
-def CreateArray3Variable(name):
-    if name in globals():
+def CreateArray3Variable(module, name):
```
Comment (adityaghantasala) [inline applications/CoSimulationApplication/custom_data_structure/pyKratos/Variable.py:62]: What does this module do ?
you are not using it here right ?

**Review by adityaghantasala — APPROVED** [review]

File: applications/CoSimulationApplication/custom_data_structure/pyKratos/Variable.py (hunk 1 above)
Comment (philbucher) [reply applications/CoSimulationApplication/custom_data_structure/pyKratos/Variable.py:62]: it is used, in line 68

Comment (philbucher) [reply applications/CoSimulationApplication/custom_data_structure/pyKratos/Variable.py:62]: fyi this is necessary for having "application" variables => e.g. `KratosMultiphysics.TIME` or `KratosMultiphysics.CoSimulationApplication.MY_COSIM_VAR`

**Pull request merged**